Refactor: #159 폴더/파일 정리 및 네이밍 일관화 작업

MuchanKim opened the pull request:
### 📌 관련 이슈
- Closes #159

---

### 🤔 Why (리팩토링 이유)

- 역할별 폴더가 혼재되어 있고, 네이밍/위치 일관성이 부족해 코드 탐색과 관리가 어려웠음

---

### 📝 What (주요 변경 내용)

- Manager 클래스 Manager 폴더로 이동
- Presentation 폴더 내 각 페이지별로 View/ViewModel 폴더 분리 및 파일 이동
- 팝오버 뷰 관련 파일을 해당 화면(View) 폴더로 이동
- Protocol 관련 파일 Sources/Protocols 폴더로 통합 및 Detector 프로토콜 분리
- SwiftDataUserTrainingDataRepositoy.swift 파일명 오타 수정
- 폴더 네이밍 복수형(Models, Protocols, Services 등)으로 통일

---

### 🗂️ 변경된 구조 (Before → After)

#### 1. Manager/Service

```
[Before]
Sources/
  Service/
    ActualTimerManager.swift
    VirtualTimerManager.swift
    FaceAbsenceTimer.swift
    VisionBlinkDetector.swift
    VisionYawnDetector.swift

[After]
Sources/
  Manager/
    ActualTimerManager.swift
    VirtualTimerManager.swift
  Services/
    FaceAbsenceTimer.swift
    VisionBlinkDetector.swift
    VisionYawnDetector.swift
```

---

#### 2. Presentation/FeedbackView & ReportView

```
[Before] 
Sources/
  Presentation/
    FeedbackView/
      FeedbackView.swift
      FeedbackViewModel.swift
      HoverRatingView.swift
    ReportView/
      ReportView.swift
      ReportViewModel.swift
      TimerReportBox.swift

[After]
Sources/
  Presentation/
    FeedbackView/
      View/
        FeedbackView.swift
        HoverRatingView.swift
      ViewModel/
        FeedbackViewModel.swift
    ReportView/
      View/
        ReportView.swift
        TimerReportBox.swift
      ViewModel/
        ReportViewModel.swift
```

---

#### 3. Protocol/Protocols 폴더 및 Detector 분리

```
[Before]
Sources/
  Protocol/
    UserTrainingDataRepository.swift
  ML/
    Protocol/
      Predictor.swift
      Personalizater.swift
  Service/
    VisionBlinkDetector.swift  (내부에 BlinkDetector 프로토콜)
    VisionYawnDetector.swift   (내부에 YawnDetector 프로토콜)

[After]
Sources/
  Protocols/
    UserTrainingDataRepository.swift
    Predictor.swift
    Personalizater.swift
    Detector.swift   // (BlinkDetector, YawnDetector 프로토콜 분리)
  Services/
    VisionBlinkDetector.swift
    VisionYawnDetector.swift
```

---

#### 4. 파일명 오타 수정

```
[Before]
Sources/
  Data/
    SwiftDataUserTrainingDataRepositoy.swift

[After]
Sources/
  Data/
    SwiftDataUserTrainingDataRepository.swift
```
---


### ✅ 체크리스트

- [x] 관련 이슈를 닫는 커밋 메시지 사용 (`Closes #159`)
- [x] 동작 확인 완료 (시뮬레이터 / 실기기)
- [x] PR 제목, 설명 명확히 작성됨
- [x] 커밋 메시지 규칙 준수
- [x] 불필요한 주석/코드 제거

---

### 💬 리뷰 포인트

- 폴더/파일 이동으로 인한 문제 발생 여부, 빌드 시 문제 없었습니다! 그래도 확인 부탁드려요.
- PopOverView폴더를 각 뷰 폴더로 분리해서 넣을까 고민했는데 현재 작업단위에선 하지 않았습니다.

MuchanKim committed:
chore: #159 SwiftDataUserTrainingDataRepositoy.swift 파일명 오타 수정

why
파일명 오타를 바로잡아 검색성과 가독성을 높이기 위함

what
SwiftDataUserTrainingDataRepositoy.swift를 SwiftDataUserTrainingDataRepository.swift로 변경

MuchanKim committed:
chore: #159 ML 폴더 내 Predictor/Personalizater 프로토콜 파일을 Protocols 폴더로 이동

why
프로토콜 파일 위치를 통합해 관리 및 검색성을 높이기 위함

what
ML 폴더에 있던 Predictor.swift, Personalizater.swift를 Sources/Protocols 폴더로 이동

MuchanKim committed:
chore: #159 Protocol 폴더명을 Protocols로 변경

why
폴더명 복수형 통일로 관리 및 탐색성 향상

what
Protocol -> Protocols 이름 변경

MuchanKim committed:
chore: #159 Services 폴더명 복수형으로 변경

why
폴더명 복수형 통일로 프로젝트 구조 일관성 강화

what
Service 폴더를 Services로 이름 변경 및 관련 import 경로 수정

MuchanKim committed:
chore: #159 Service 폴더 내 매니저 클래스 Manager 폴더로 이동

why
역할별로 폴더 구조를 명확히 하여 코드 가독성과 유지보수성을 높이기 위함

what
Service 폴더에 있던 ActualTimerManager.swift, VirtualTimerManager.swift, AnalysisManager.swift, CameraManager.swift 파일을 Manager 폴더로 이동

MuchanKim committed:
chore: #159 BlinkDetector, YawnDetector 프로토콜 파일 분리

why
프로토콜과 구현부를 분리하여 역할을 명확히 하고, 코드 재사용성과 관리 편의성을 높이기 위함

what
BlinkDetector, YawnDetector 프로토콜을 기존 코드 구현 파일에서 분리하여 별도의 프로토콜 파일로 이동

MuchanKim committed:
chore: #159 피드백뷰/리포트뷰 폴더 내 파일 View, ViewModel 폴더로 분리

why  
화면별로 View와 ViewModel을 분리해 폴더 구조 일관성과 유지보수성을 높이기 위함

what  
FeedbackView, FeedbackViewModel 등 피드백뷰 관련 파일과  
ReportView, ReportViewModel 등 리포트뷰 관련 파일을  
각각 View/와 ViewModel/ 폴더로 이동

bisor0627 commented on NoMyeolmang/NoMyeolmang/Sources/Presentation/FeedbackView/View/FeedbackView.swift:
디렉토리 구조를 보면 View 단위(`FeedbackView`, `ReportView`) 아래에 `View`, `ViewModel` 등의  따로 나누셨는데, 이는 역할 기반 분리를 하신 걸로 봐도 될까요?

현재 `FeedbackView/View/FeedbackView.swift`와 같이 디렉토리와 파일 명 모두에 "View"가 중복되는 것이 보입니당

디렉토리 이름에 `View`를 유지하기보다, `Presentation/Feedback/FeedbackView.swift`처럼 의미 있는 feature 명만 디렉토리로 유지하고, View/Model 등은 파일명으로 책임을 명확히 표현하는 방식이 더 명료할 수 있지 않을까 살짝 리뷰 남기고 갑니다

bisor0627 replied:
또한... `Image("backgroundSpace")`의 경우 현재는 `camelCase` 형식인데요!  
[여기](https://github.com/DeveloperAcademy-POSTECH/2025-C4-M9-NoMyeolmang/blob/29e83f261e0699dca3601707324c4f15308bb976/NoMyeolmang/NoMyeolmang/Sources/Presentation/FeedbackView/View/FeedbackView.swift#L16)

이미지 asset 이름은 일반적으로 `snake_case`(소문자 + 언더스코어)를 사용하는 것이 권장된다고 하여,  
예: `background_space`  
관련된 [이 글](https://21zerixpm.medium.com/the-correct-asset-naming-every-ios-developer-should-use-01b564edce6d)을 공유하고... 슬쩍 지나가봅니다

MuchanKim replied:
오호 그렇네요! FeedBackView보다는 Feedback에서 view/viewmodel으로 분기하는게 더 명료한 것 같습니다! 👍

MuchanKim replied:
> 디렉토리 구조를 보면 View 단위(FeedbackView, ReportView) 아래에 View, ViewModel 등의 따로 나누셨는데, 이는 역할 기반 분리를 하신 걸로 봐도 될까요?

네! 화면단위로 나눠서 뷰와 뷰모델로 구분하였어요. 

> 이미지 asset 이름은 일반적으로 snake_case(소문자 + 언더스코어)를 사용하는 것이 권장된다고 하여, 예: background_space

오우 감사합니다. 너무 잘 봐주시는데요? 에셋은 스네이크 메모..

ohdodin replied:
> 또한... `Image("backgroundSpace")`의 경우 현재는 `camelCase` 형식인데요! [여기](https://github.com/DeveloperAcademy-POSTECH/2025-C4-M9-NoMyeolmang/blob/29e83f261e0699dca3601707324c4f15308bb976/NoMyeolmang/NoMyeolmang/Sources/Presentation/FeedbackView/View/FeedbackView.swift#L16)
> 
> 이미지 asset 이름은 일반적으로 `snake_case`(소문자 + 언더스코어)를 사용하는 것이 권장된다고 하여, 예: `background_space` 관련된 [이 글](https://21zerixpm.medium.com/the-correct-asset-naming-every-ios-developer-should-use-01b564edce6d)을 공유하고... 슬쩍 지나가봅니다

오홍 새로운 좋은 정보 알고갑니다! 감사해요 커비👍

bisor0627 commented:
더불어... 아래와 같은 하드코딩된 값들은 기억이 아직 남아계실 때
이후 유지보수를 위해 상수화해서 사용하시면 좋겠다는 생각이 들었습니다..
- [`private var goalTime: Int = 300`](https://github.com/DeveloperAcademy-POSTECH/2025-C4-M9-NoMyeolmang/blob/29e83f261e0699dca3601707324c4f15308bb976/NoMyeolmang/NoMyeolmang/Sources/Manager/ActualTimerManager.swift#L19)  
- [`Color("ReportViewSelectedColor")` vs. `Color("5C436D")`](https://github.com/DeveloperAcademy-POSTECH/2025-C4-M9-NoMyeolmang/blob/29e83f261e0699dca3601707324c4f15308bb976/NoMyeolmang/NoMyeolmang/Sources/Presentation/FeedbackView/View/HoverRatingView.swift#L32)

그리고 아래와 같이 동작에 민감한 threshold의 경우에는  
꼭 외부로 extract하지 않더라도 이 수치가 어떤 기준인지 짧게라도 설명이 있으면 더 이해하기 쉬울 것 같다는 생각을 했습니다!

- [`private let earThreshold: CGFloat = 0.18`](https://github.com/DeveloperAcademy-POSTECH/2025-C4-M9-NoMyeolmang/blob/29e83f261e0699dca3601707324c4f15308bb976/NoMyeolmang/NoMyeolmang/Sources/Services/VisionBlinkDetector.swift#L11-L12)  
- [`private let marThreshold: CGFloat = 1.6`](https://github.com/DeveloperAcademy-POSTECH/2025-C4-M9-NoMyeolmang/blob/29e83f261e0699dca3601707324c4f15308bb976/NoMyeolmang/NoMyeolmang/Sources/Services/VisionYawnDetector.swift#L11-L12)

MuchanKim committed:
chore: #159 Presentation 폴더 내 각 화면 폴더명에서 ‘View’ 접미사 제거

why
폴더명에 ‘View’ 접미사가 포함되어 디렉터리와 파일명 중복이 발생하고, 역할이 불명확해 보여 접미사를 제거함

what
Presentation/FeedbackView → Presentation/Feedback
Presentation/ReportView → Presentation/Report
Presentation/TimerView → Presentation/Timer
Presentation/TimerSettingView → Presentation/TimerSetting
각각 폴더명을 변경하고 내부에 View, ViewModel 폴더로 역할 기반 분리 적용

MuchanKim commented:
> 더불어... 아래와 같은 하드코딩된 값들은 기억이 아직 남아계실 때 이후 유지보수를 위해 상수화해서 사용하시면 좋겠다는 생각이 들었습니다..

맞습니다.. 매직넘버들도 처리해야합니다. 이런 것들이 한 둘이아니라 이슈를 파서 한번에 처리하면 좋을 것 같네요!

> 그리고 아래와 같이 동작에 민감한 threshold의 경우에는 꼭 외부로 extract하지 않더라도 이 수치가 어떤 기준인지 짧게라도 설명이 있으면 더 이해하기 쉬울 것 같다는 생각을 했습니다!

DocC 작업할 때 참고하겠습니다.! 이 안건도 별도의 작업 단위에서 다루면 좋을 것 같네요 후후.

정성 리뷰 감사합니다👍

ohdodin reviewed (approved):
확인 완료입니다!
